base_domain_operator/expression.py

```diff
@@ -0,0 +1,62 @@
+# -*- coding: utf-8 -*-
+# © 2017 Therp BV <http://therp.nl>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+from openerp import api
+# pylint: disable=W0402
+from openerp.osv.expression import ExtendedLeaf, expression, is_operator,\
+    is_leaf
+
+
+class BaseDomainOperatorExtendedLeaf(ExtendedLeaf):
+    def __init__(self, leaf, model, join_context=None, internal=False):
+        if isinstance(leaf, ExtendedLeaf):
+            # TODO: think very long and hard if this introduces a security
+            # problem
+            internal = True
+            model = leaf.model
+            join_context = (join_context or []) + leaf.join_context
+            leaf = leaf.leaf
+        super(BaseDomainOperatorExtendedLeaf, self).__init__(
+            leaf, model, join_context=join_context, internal=internal
+        )
+
+
+class Expression(expression):
+    def parse(self, cr, uid, context):  # pylint: disable=R8110
+        domain = []
+        env = api.Environment(cr, uid, context)
```

Comment:
this environment is use in the next line already, and we definitely need one for the first operator check, so this wouldn't really save anything